app/controllers/mailing_list_signups_controller.rb

```diff
@@ -0,0 +1,89 @@
+class MailingListSignupsController < ApplicationController
+  include CircuitBreaker
+  layout "registration"
+
+  SESSION_STORE_KEY = :mailing_list_signup
+
+  def new
+    @signup = MailingList::Signup.new
+  end
+
+  def create
+    @signup = MailingList::Signup.new(new_signup_params)
+
+    if @signup.valid?
+      save_signup_to_session
+
+      return redirect_to(edit_mailing_list_signup_path) if @signup.already_signed_up?
```

Comment:
This is a bit misleading I think - `already_signed_up?` will return `true` if the candidate already exists, not if they exist _and_ they have already signed up to the mailing list

Reply:
Yeah - I named it when I didn't fully understand what was going on. Renamed to `#exists_in_crm?`.

Reply:
That kind of reads like the same thing when you call it on `@signup.exists_in_crm?` 😬 maybe something like `@signup.existing_candidate?`